abstractions/dotnet/src/authentication/IAuthenticationProvider.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Kiota.Abstractions.Authentication {
+    /// <summary>
+    /// Authenticates the application request.
+    /// </summary>
+    public interface IAuthenticationProvider {
+        /// <summary>
+        /// Authenticates the application request.
+        /// </summary>
+        /// <param name="request">The request to authenticate.</param>
+        /// <returns>A task to await for the authentication to be completed.</returns>
+        Task AuthenticateRequest(RequestInfo request);
```

Comment:
Since we kind of know that the implementers will most likely be async maybe we should add the `Async` suffix so that it becomes `AuthenticateRequestAsync`?

Reply:
Thanks, very good point. Hoping between languages I kind of forgot about this convention in dotnet. Fixed